### PR TITLE
[ONNX] Support opset13 Squeeze and Unsqueeze

### PR DESCRIPTION
**Pull request by BowenBao**
In opset 13, squeeze/unsqueeze is updated to take `axes` as input, instead of attribute.

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/shape_type_inference.cpp [hunk 1]
```diff
@@ -223,7 +226,15 @@ Value* CloneValueFromListConstruct(Value* v, std::shared_ptr<Graph> n_graph) {
       auto new_input = n_graph->addInput();
       new_input->copyMetadata(input);
       unsqueezed_node->addInput(new_input);
-      unsqueezed_node->is_(attr::axes, {0});
+      if (opset_version >= 13) {
```
Comment (fatcat-z) [inline torch/csrc/jit/passes/onnx/shape_type_inference.cpp:229]: nit: Would it be better that we create a new method with a meaningful name to increase readability and let people know why we have these new code? Like, change this line to "if (unsqueezeNeedsMultipleParameters(opset_version))". The method is simple as:

bool unsqueezeNeedsMultipleParameters(int opset_version) {
    return opset_version >= 13;
}

We can put it into helper.h file so this method could be shared among different places, like shape_type_inference.cpp and peehole.cpp.

File: torch/csrc/jit/passes/onnx/peephole.cpp [hunk 2]
```diff
@@ -442,7 +450,15 @@ void fixDefaultRNNState(
   Node* unsqueezed_num_directions = graph->create(onnx::Unsqueeze, 1);
   unsqueezed_num_directions->insertBefore(n);
   unsqueezed_num_directions->addInput(num_directions->outputs()[0]);
-  unsqueezed_num_directions->is_(attr::axes, {0});
+  if (opset_version >= OPSET_VERSION_13) {
```
Comment (fatcat-z) [inline torch/csrc/jit/passes/onnx/peephole.cpp:453]: nit: Would it be better that we create a new method with a meaningful name to increase readability and let people know why we have these new code? Like, change this line to "if (unsqueezeNeedsMultipleParameters(opset_version))". The method is simple as:

bool unsqueezeNeedsMultipleParameters(int opset_version) {
    return opset_version >= 13;
}

We can put it into helper.h file so this method could be shared among different places, like shape_type_inference.cpp and peehole.cpp.

File: torch/csrc/jit/passes/onnx/peephole.cpp [hunk 3]
```diff
@@ -418,7 +418,15 @@ void fixDefaultRNNState(
   Node* unsqueezed_batch_size = graph->create(onnx::Unsqueeze, 1);
   unsqueezed_batch_size->insertBefore(n);
   unsqueezed_batch_size->addInput(batch_size->outputs()[0]);
-  unsqueezed_batch_size->is_(attr::axes, {0});
+  if (opset_version >= OPSET_VERSION_13) {
```
Comment (fatcat-z) [inline torch/csrc/jit/passes/onnx/peephole.cpp:421]: nit: Would it be better that we create a new method with a meaningful name to increase readability and let people know why we have these new code? Like, change this line to "if (unsqueezeNeedsMultipleParameters(opset_version))". The method is simple as:

bool unsqueezeNeedsMultipleParameters(int opset_version) {
    return opset_version >= 13;
}

We can put it into helper.h file so this method could be shared among different places, like shape_type_inference.cpp and peehole.cpp.

In addition, for this block of code(line 422 - 428), given they are repeated in different places. Should we also consider create a common method to hold them?

Comment (BowenBao) [reply torch/csrc/jit/passes/onnx/peephole.cpp]: Thanks for the suggestion. I'm creating a common helper function for creating onnx unsqueeze node. And adding comments explaining why the difference from opset version.

**Review by fatcat-z — APPROVED** [review]

File: torch/csrc/jit/passes/onnx/helper.cpp [hunk 4]
```diff
@@ -96,5 +96,28 @@ Value* addInputToBlock(Block* block) {
   return block->addInput();
 }
 
+Node* createONNXUnsqueeze(
+    std::shared_ptr<Graph> graph,
```
Comment (neginraoof) [inline torch/csrc/jit/passes/onnx/helper.cpp:100]: Maybe change this to Graph* to resolve the build failure.

File: torch/onnx/symbolic_opset9.py [hunk 5]
```diff
@@ -186,7 +186,7 @@ def cat(g, tensor_list, dim):
 
 @parse_args('v', 'i')
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset9.py:187]: Please also update the code in squeeze/unsqueeze symbolics to use the helper function.

File: torch/csrc/jit/passes/onnx/helper.cpp (hunk 4 above)
Comment (BowenBao) [reply torch/csrc/jit/passes/onnx/helper.cpp]: thanks for catching, updated.

**Review by neginraoof — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**